packages/protocol/test/governance/network/governance.ts

```diff
@@ -1591,6 +1591,52 @@ contract('Governance', (accounts: string[]) => {
       await assertRevert(governance.vote(otherProposalId, index, value))
     })
 
+    describe('when voting on two proposals', () => {
+      const proposalId1 = 2
+      const proposalId2 = 3
+      const index1 = 1
+      const index2 = 2
+      beforeEach(async () => {
+        const newDequeueFrequency = 60
+        await governance.setDequeueFrequency(newDequeueFrequency)
+        await governance.propose(
+          [transactionSuccess1.value],
+          [transactionSuccess1.destination],
+          // @ts-ignore bytes type
+          transactionSuccess1.data,
+          [transactionSuccess1.data.length],
+          descriptionUrl,
+          // @ts-ignore: TODO(mcortesi) fix typings for TransactionDetails
+          { value: minDeposit }
+        )
+        await timeTravel(newDequeueFrequency, web3)
+        await governance.approve(proposalId1, index1)
+        const stage1 = await governance.getProposalStage(proposalId1)
+        console.log(stage1.toNumber())
+        await governance.propose(
+          [transactionSuccess2.value],
+          [transactionSuccess2.destination],
+          // @ts-ignore bytes type
+          transactionSuccess2.data,
+          [transactionSuccess2.data.length],
+          descriptionUrl,
+          // @ts-ignore: TODO(mcortesi) fix typings for TransactionDetails
+          { value: minDeposit }
+        )
+        await timeTravel(newDequeueFrequency, web3)
+        await governance.approve(proposalId2, index2)
+        await timeTravel(approvalStageDuration, web3)
+        await mockLockedGold.setAccountTotalLockedGold(account, weight)
+      })
+
+      it('should set mostRecentReferendumProposal to the youngest proposal voted on', async () => {
```

Comment:
What do you think about adding an actual revert expectation and a call to `isVoting`?

Reply:
The current test feels like the correct minimal unit test for the exploit scenario.

Reply:
That assumes though that `isVoting` calls `mostRecentReferendumProposal` right? IMO, it seems easy to do and gives us the actual invariant that consumers of the `Governance` API care about?